pkg/vendir/cmd/sync.go

```diff
@@ -50,6 +51,7 @@ func NewSyncCmd(o *SyncOptions) *cobra.Command {
 
 	cmd.Flags().StringSliceVarP(&o.Directories, "directory", "d", nil, "Sync specific directory (format: dir/sub-dir[=local-dir])")
 	cmd.Flags().BoolVarP(&o.Locked, "locked", "l", false, "Consult lock file to pull exact references (e.g. use git sha instead of branch name)")
+	cmd.Flags().BoolVar(&o.Lazy, "lazy", true, "Override \"lazy\" value in directory content configuration")
```

Comment:
Maybe we can change the text here to something like:
"When this flag is set to 'false' it ignores the 'lazy' value in the directory content configuration"
Do you think it is easier to understand what the flag does?

Reply:
I was thinking along the same lines. It it less vague about the purpose of this option.

Reply:
Hi @joaopapereira  This CLI option 'lazy' (default value is true) is really confusing. With vendir 0.37.0, `vendir sync --lazy=true` won't do lazy fetch.  This behavior does not comply with the proposal:
https://github.com/carvel-dev/carvel/tree/develop/proposals/vendir/001-lazy-synching-on-stable-config#other-approaches-considered
![图片](https://github.com/carvel-dev/vendir/assets/10006/4487823a-81a4-4e80-9d87-311bd99956f6)

This proposal should mean the CLI option --lazy default to **false**, and if --lazy is set to true, vendir will do lazy fetch,
Also this PR defines 'Lazy bool' as a bool instead of `*bool`, so it can not be used to override the CLI option '--lazy' because it always has a default value 'false'.
```
type DirectoryContents struct {
	Path string `json:"path"`
	Lazy bool   `json:"lazy,omitempty"`
```

Reply:
@jessehu The proposal was:

- to add the `lazy` option to `vendir.yml` to allow skipping sync when the config is not changed,
- to add the `--lazy` CLI argument to temporarily disable the effect of the `lazy` option.

What you reference to is a _considered_ approach that could've satisfied the use case but wasn't accepted.

![image](https://github.com/carvel-dev/vendir/assets/1475583/24a51beb-9570-4f6e-872e-b6bcd0b1ab39)

Reply:
I see. Thanks. However this 'other approach considered' makes more sense. The current code behaves a little weird: 

1. vendir sync --lazy=true won't do lazy fetch (--lazy default to true) since it **does not overrites** in lazy setting in vendir.yml
2. vendir sync --lazy=false won't do lazy fetch since it **overrites** in lazy setting in vendir.yml

According to this behaviour, this CLI flag should be named `ignoreLazySetting` or `noLazy` rather than `lazy` (the setting name in verdir.yml)
```
cmd.Flags().BoolVar(&o.Lazy, "lazy", true, "Set to 'false' it ignores the 'lazy' flag in the directory content configuration")
```

Reply:
The alternative approach is less flexible, as it doesn't allow configuring "laziness" per directory in the `vendir.yml` configuration file.

The selected name of the CLI flag is a little confusing, I agree. It might've been more explicit to use, as you mentioned, `--noLazy` or alike instead of `--lazy=false`.